application/helpers/frontend_helper.php

```diff
@@ -1865,6 +1865,10 @@ function doAssessment($surveyid, $returndataonly=false)
             $groups=array();
             foreach($fieldmap as $field)
             {
+                // Init Assessment Value
+                $assessmentValue = NULL;
```

Comment:
I was reviewing code. This code is inside doAssessment function.
If we only want this to execute when assessment is enabled, I guess the best is to look for those places and check the settings there.
Example: 
- replacements_helper.php around line 545

It is infact cached on SurveyRuntimeHelper.php around line 708, 757 and 783